### PR TITLE
Temporary PR to show whats changed from the Raytraced AO stuff

### PR DESCRIPTION
**Pull request by devshgraphicsprogramming**

### CODE REVIEW

File: include/irr/builtin/glsl/bxdf/common.glsl [hunk 1]
```diff
@@ -135,7 +135,7 @@ irr_glsl_IsotropicViewSurfaceInteraction irr_glsl_calcFragmentShaderSurfaceInter
 {
     irr_glsl_IsotropicViewSurfaceInteraction interaction;
     interaction.V.dir = _View;
-    interaction.V.dPosdScreen[0] = dFdx(_SurfacePos);
+    interaction.V.dPosdScreen[0] = dFdx(_SurfacePos);// TODO: require the user to feed us the derivatives
```
Comment (devshgraphicsprogramming) [inline include/irr/builtin/glsl/bxdf/common.glsl:138]: @Crisspl this is a problem when including the file in other shaders than fragment shader

File: src/irr/asset/CMeshManipulator.cpp [hunk 2]
```diff
@@ -868,7 +869,7 @@ void CMeshManipulator::copyMeshBufferMemberVars<ICPUMeshBuffer>(ICPUMeshBuffer*
     auto ixBinding = _src->getIndexBufferBinding()[0];
 	_dst->setIndexBufferBinding(
 		std::move(ixBinding)
-	);/*
+	);/* @Crisspl wtf with the commenting out???
```
Comment (Crisspl) [inline src/irr/asset/CMeshManipulator.cpp:872]: see `createMeshBufferDuplicate`, also note taht `copyMeshBufferMemberVars` is private

File: src/irr/ext/MitsubaLoader/CMitsubaLoader.cpp [hunk 3]
```diff
@@ -686,7 +693,7 @@ core::smart_refctd_ptr<asset::ICPUPipelineLayout> CMitsubaLoader::createPipeline
 	pcrng.size = sizeof(uint32_t);//instance data offset
 	pcrng.stageFlags = static_cast<asset::ISpecializedShader::E_SHADER_STAGE>(asset::ISpecializedShader::ESS_FRAGMENT | asset::ISpecializedShader::ESS_VERTEX);
 
-	core::smart_refctd_ptr<ICPUDescriptorSetLayout> ds0layout;
+	core::smart_refctd_ptr<ICPUDescriptorSetLayout> ds0layout; // needs to builtin and cached statically
```
Comment (Crisspl) [inline src/irr/ext/MitsubaLoader/CMitsubaLoader.cpp:696]: cant because of VT

Comment (devshgraphicsprogramming) [reply src/irr/ext/MitsubaLoader/CMitsubaLoader.cpp:696]: yeah, forgot to remove the comment... you can remove it, I guess.

**Pull request merged**